add note about multiple deployments in same k8s cluster (#662)
temporalio/documentation: docs/server/production-deploy.md

See the [versions & dependencies page](/docs/server/versions-and-dependencies/) for precise versions we support together with these features.

> Kubernetes is not required for Temporal, but it is popular anyway. We do maintain [a Helm chart](https://github.com/temporalio/helm-charts) as a reference and we recently hosted a YouTube discussion on how we think about the Kubernetes ecosystem in relation to Temporal.

Reviewer: Maybe link to the youtube discussion?

Reviewer 2: ```suggestion
Kubernetes is not required for Temporal, but it is popular anyway. We do maintain [a Helm chart](https://github.com/temporalio/helm-charts) as a reference. We also [hosted a YouTube discussion](https://www.youtube.com/watch?v=11I87HKS_NM) on how we think about the Kubernetes ecosystem in relation to Temporal.
```

### Configuration

At minimum, the `development.yaml` file needs to have the [`global`](/docs/server/configuration/#global) and [`persistence`](https://docs.temporal.io/docs/server/configuration/#persistence) parameters defined.

The [Server configuration reference](/docs/server/configuration) has a more complete list of possible parameters.

> ### Before You Deploy: A Note on Shard Count

Reviewer 2: ```suggestion
### Before you deploy: Reminder on shard count
```

This is why we recommend tracking `ScheduleToStart` latency for determining if the task queue has a backlog (aka Workers are under-provisioned for a given Task Queue).
We do plan to add features that give more visibility into the task queue state in future.

### FAQ: High Availability cluster configuration

You can set up a high availability deployment by running more than one instance of the server. Temporal also handles [membership and routing](https://docs.temporal.io/blog/workflow-engine-principles/#membership-and-routing-1350). You can find more details in [the `clusterMetadata` section of the Server Configuration reference](https://docs.temporal.io/docs/server/configuration/#clustermetadata).

```yaml
clusterMetadata:
  enableGlobalNamespace: false
  failoverVersionIncrement: 10
  masterClusterName: "active"
  currentClusterName: "active"
  clusterInformation:
    active:
      enabled: true
      initialFailoverVersion: 0
      rpcAddress: "127.0.0.1:7233"
```

> ### FAQ: Multiple deployments on a single cluster
> 
> If you are planning to have multiple temporal deployments within the same k8s cluster, doublecheck the following:
> 
> - Have a separate persistence (database) for each deployment
> - Cluster membership ports should be different for each deployment. For example: 
>   - Temporal1 services can have 7233 for frontend, 7234 for history, 7235 for matching
>   - temporal2 services can have 8233 for frontend, 8234 for history, 8235 for matching
> 
> [More details about the reason here](https://github.com/temporalio/temporal/issues/1234).

Reviewer 2: ```suggestion
### FAQ: Multiple deployments on a single cluster

You may sometimes want to have multiple parallel deployments on the same cluster, eg:
- when you want to split Temporal deployments based on namespaces, e.g. staging/dev/uat, or for different teams who need to share infrastructure
- when you need a new deployment to change `NUM_HISTORY_SHARDS`

If you are planning to have multiple temporal deployments within the same k8s cluster, double-check the following:

- Have a separate persistence (database) for each deployment
- Cluster membership ports should be different for each deployment (they can be set through environment variables). For example: 
  - Temporal1 services can have 7233 for frontend, 7234 for history, 7235 for matching
  - Temporal2 services can have 8233 for frontend, 8234 for history, 8235 for matching
- There is no need to change gRPC ports.

[More details about the reason here](https://github.com/temporalio/temporal/issues/1234).
```

```yaml
clusterMetadata:
  enableGlobalNamespace: false
  failoverVersionIncrement: 10
  masterClusterName: "active"
  currentClusterName: "active"
  clusterInformation:
    active:
      enabled: true
      initialFailoverVersion: 0
      rpcAddress: "127.0.0.1:7233"
```

### FAQ: Multiple deployments on a single cluster

You may sometimes want to have multiple parallel deployments on the same cluster, eg:

- when you want to split Temporal deployments based on namespaces, e.g. staging/dev/uat, or for different teams who need to share common infrastructure.
- when you need a new deployment to change `numHistoryShards`.

**We recommend not doing this if you can avoid it**. If you need to do it anyway, double-check the following:

- Have a separate persistence (database) for each deployment
- Cluster membership ports should be different for each deployment (they can be set through environment variables). For example:
  - Temporal1 services can have 7233 for frontend, 7234 for history, 7235 for matching
  - Temporal2 services can have 8233 for frontend, 8234 for history, 8235 for matching
- There is no need to change gRPC ports.

> [More details about the reason here](https://github.com/temporalio/temporal/issues/1234).

Reviewer 2: @meetchandan last call for anything else you wanna add before we merge this update! ty for feedback so far
